Add spike extension

data-pirate opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

- Maybe you should look into useCachedPromise/usePromise instead of useEffects

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

data-pirate commented:
Hi 👋

Thank you so much for your valuable feedback!

I've made the requested updates:

- Added screenshots of the extension for better visualization.
- Replaced useEffect with useCachedPromise for improved caching and performance.

Thank you again for your time and support!

pernielsentikaer commented:
Could you look into the [CI errors](https://github.com/raycast/extensions/actions/runs/11563511653/job/32186954462?pr=14968) mentioned?

data-pirate commented:
Thank you so much for the review. I have fixed the CI issues.

pernielsentikaer commented on extensions/spike/src/openIncidents.tsx:
```diff
@@ -0,0 +1,126 @@
+    <MenuBarExtra isLoading={isLoading} icon="spike-logo-white.png" tooltip="Open incidents">
+      <MenuBarExtra.Item
+        icon={{
+          source: isOnCall ? "green-dot.png" : "gray-dot.png",
```
You can also use `{ source: Icon. Dot, tintColor: Color.Green }`

data-pirate replied:
Hi there, i have updated the code to use `Icon.Dot` instead of the images. i have also deleted the unused images.

pernielsentikaer reviewed (commented):
You can reference images from metadata directly in your README, no reason to add it from `assets`

data-pirate commented:
Hi there,

I've noticed that the following error has started appearing again in CI pipeline:

```Entering /Users/runner/work/extensions/extensions/extensions/spike/
Error: We are restricting public organisation extensions for the moment. Ping @pedro on Slack to discuss it. (spike)
Error: Process completed with exit code 1
```

This issue was previously fixed, but seems to have resurfaced.

pernielsentikaer commented:
That's because for me to test it I need to only have one folder in the PR 🙂

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
